### PR TITLE
Persist inventory through multiple client sessions

### PR DESCRIPTION
**Pull request by Shirkie01**
Changes the Player Id from a ushort to a NitroxId, and removes the PlayerGameObjectId.

### CODE REVIEW

**Comment by Sunrunner37** [comment]
Is this something that we want to do?  It effectively increases every packet size that contains a player id from 16 bits to 16 bytes.

**Comment by Shirkie01** [comment]
I was working on getting inventory to persist across logins. Right now a new Player.GameObjectId is created every time a player joins so their stuff is lost. Perhaps instead of changing everything to use NitroxId, I can try including the GameObjectId as part of the PersistedPlayerData.

File: NitroxModel/DataStructures/NitroxId.cs [hunk 1]
```diff
@@ -52,5 +52,15 @@ public override string ToString()
...
+
+        public static bool operator ==(NitroxId id1, NitroxId id2)
+        {
+            return id1?.guid == id2?.guid;
+        }
+
+        public static bool operator !=(NitroxId id1, NitroxId id2)
+        {
+            return !(id1 == id2);
+        }
```
Comment (Coding-Hen) [inline NitroxModel/DataStructures/NitroxId.cs]: Is this needed anymore as you are not using/changing the NitroxId anywhere else?

File: NitroxServer/Player.cs [hunk 2]
```diff
@@ -23,13 +23,13 @@ public class Player : IProcessorContext
...
 
-        public Player(PlayerContext playerContext, NitroxConnection connection, Vector3 position, Optional<NitroxId> subRootId)
+        public Player(PlayerContext playerContext, NitroxConnection connection, Vector3 position, Optional<NitroxId> subRootId, NitroxId gameObjectId)
         {
             PlayerContext = playerContext;
             this.connection = connection;
             Position = position;
             SubRootId = subRootId;
-            GameObjectId = new NitroxId();
+            GameObjectId = gameObjectId;
```
Comment (Sunrunner37) [inline NitroxServer/Player.cs]: Do we need to have the overhead of persisting this data?  Can we make it deterministic.... maybe based on the player name?  I believe we used to do that but it fell off at some point in time.

File: NitroxModel/DataStructures/NitroxId.cs (hunk 1 above)
Comment (Shirkie01) [reply NitroxModel/DataStructures/NitroxId.cs]: The new direction is to use a deterministic NitroxId, so I am changing the class now. I would like to keep the operators as there are lots of places in the code where == is used to check equality.

File: NitroxServer/Player.cs (hunk 2 above)
Comment (Shirkie01) [reply NitroxServer/Player.cs]: I've changed the code to use a deterministic NitroxId following the RFC guidelines for GUID version 5.

**Comment by Shirkie01** [comment]
Closing as apparently we don't care about "the overhead of persisting this data" and "making the NitroxId deterministic based on the player name" any more as seen in pull request #902.

**Pull request closed without merging**